src/baseline_rule.py

```diff
@@ -29,22 +30,28 @@ class BaselineRule:
     """
 
     def __init__(self, rule_record):
-        self.name = rule_record.get('name', '<no name>')
-        print(f'processing rule {self.name}')
+        RuleSyntaxChecker.check_keys_legality(rule_record)
+        self.name = RuleSyntaxChecker.check_dns_subdomain_name(rule_record.get('name', 'no-name'))
         self.description = rule_record.get('description', '')
-        self.action = BaselineRuleAction[rule_record.get('action', 'allow')]
-        self.source = Selector.parse_selectors(rule_record.get('from', ''))
-        self.target = Selector.parse_selectors(rule_record.get('to', ''))
-        self.source_ns = Selector.parse_selectors(rule_record.get('from_ns', ''))
-        self.target_ns = Selector.parse_selectors(rule_record.get('to_ns', ''))
-        self.protocol = rule_record.get('protocol')
-        self.port_min = rule_record.get('port_min')
-        self.port_max = rule_record.get('port_max')
-        self.check_entries_validation()
-
-    def check_entries_validation(self):
-        if isinstance(self.source_ns, IpSelector) or isinstance(self.target_ns, IpSelector):
-            raise Exception('A namespaceSelector can not be specified in CIDR notation', self.name)
+        try:
+            self.action = \
+                BaselineRuleAction[RuleSyntaxChecker.check_action_validity(rule_record.get('action', 'allow'))]
+            self.source = \
+                RuleSyntaxChecker.check_selector_validity(Selector.parse_selectors(rule_record.get('from', '')))
+            self.target = RuleSyntaxChecker.check_selector_validity(Selector.parse_selectors(rule_record.get('to', '')))
+            self.source_ns = RuleSyntaxChecker.\
+                check_namespace_selector_validity(Selector.parse_selectors(rule_record.get('from_ns', '')))
+            self.target_ns = RuleSyntaxChecker.\
+                check_namespace_selector_validity(Selector.parse_selectors(rule_record.get('to_ns', '')))
+            self.protocol = RuleSyntaxChecker.check_protocol_validity(rule_record.get('protocol'))
+            self.port_min = RuleSyntaxChecker.check_port_validity(rule_record.get('port_min'))
+            self.port_max = RuleSyntaxChecker.check_port_validity(rule_record.get('port_max'))
+        except Exception as e:
+            raise Exception(f'{self.name} : {e}') from None
+
+        self.check_selectors_entries_combinations()
```

Comment:
```suggestion
            self.port_max = RuleSyntaxChecker.check_port_validity(rule_record.get('port_max'))
            self.check_selectors_entries_combinations()
        except Exception as e:
            raise Exception(f'{self.name} : {e}') from None
```
So we also get rule name context for selector-combination error